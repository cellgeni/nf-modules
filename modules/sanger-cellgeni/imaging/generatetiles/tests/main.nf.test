// nf-core modules test IMAGING/generatetiles
nextflow_process {

    name "Test Process IMAGING_GENERATETILES"
    script "../main.nf"
    config "./nextflow.config"
    process "IMAGING_GENERATETILES"

    tag "modules"
    tag "modules_sanger"
    tag "imaging"
    tag "imaging/generatetiles"

    test("mouse_heart - tiff") {

        when {
            process {
                """
                input[0] = [
                    [ id:'test'], // meta map
                    file(params.modules_testdata_base_path + 'imaging/tiff/mindagap.mouse_heart.wga.tiff', checkIfExists: true),
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

    test("mouse_heart - tiff - stub") {

        options "-stub"

        when {
            process {
                """
                input[0] = [
                    [ id:'test',], // meta map
                    file(params.modules_testdata_base_path + 'imaging/tiff/mindagap.mouse_heart.wga.tiff', checkIfExists: true),
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

}
