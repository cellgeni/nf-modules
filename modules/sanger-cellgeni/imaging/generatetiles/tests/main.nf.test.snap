{
    "mouse_heart - tiff - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test_tile_coords.csv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,eddcd9c793b474b87d0333a1c90a5381"
                ],
                "tile_coords": [
                    [
                        {
                            "id": "test"
                        },
                        "test_tile_coords.csv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,eddcd9c793b474b87d0333a1c90a5381"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.04.4"
        },
        "timestamp": "2025-03-18T20:57:11.833074167"
    },
    "mouse_heart - tiff": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test_tile_coords.csv:md5,736ea44566b4ff52f7bcf1d46078152b"
                    ]
                ],
                "1": [
                    "versions.yml:md5,eddcd9c793b474b87d0333a1c90a5381"
                ],
                "tile_coords": [
                    [
                        {
                            "id": "test"
                        },
                        "test_tile_coords.csv:md5,736ea44566b4ff52f7bcf1d46078152b"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,eddcd9c793b474b87d0333a1c90a5381"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.04.4"
        },
        "timestamp": "2025-03-18T20:57:04.817854347"
    }
}